src/entities/trades/curve/tokens/types.ts

```diff
@@ -23,6 +23,7 @@ export enum TokenType {
  */
 export interface CurveToken {
   isLPToken?: boolean
+  poolTokens?: () => CurveToken[]
```

Comment:
Can you elaborate on the design decision behind this? Curious about this.

Reply:
yeah so basically this is to handle the case when the token is LpToken and has underlying tokens or metatokens. So that when we have for example 3crv token like below we know what tokens are contained and we can pass it to dynamically fetched curve pools.
![Screenshot 2022-08-01 at 14 28 53](https://user-images.githubusercontent.com/33226956/182147787-db3556dc-36ed-464a-889e-7ad99364214d.png)
- And then in fetcher function when we see 3 crv pool from api it will map it to this token and put metatokens in or if its not meta just put underlying tokens in 
<img width="600" alt="Screenshot 2022-08-01 at 14 28 06" src="https://user-images.githubusercontent.com/33226956/182147680-190777f1-82bb-4456-b1f3-897764ac6876.png">

- Actually current implementation can be switched so we entirely rely on the api for fetching all pools without having to hardcode them. We would just need to adjust it if they come with some pools with new abi-s which would break current implementation as well...